addons/event_sale/models/sale_order_line.py

```diff
@@ -11,11 +11,17 @@ class SaleOrderLine(models.Model):
         compute="_compute_event_id", store=True, readonly=False, precompute=True,
         help="Choose an event and it will automatically create a registration for this event.")
     event_ticket_id = fields.Many2one(
-        'event.event.ticket', string='Event Ticket',
+        'event.event.ticket', string='Ticket Type',
         compute="_compute_event_ticket_id", store=True, readonly=False, precompute=True,
         help="Choose an event ticket and it will automatically create a registration for this event ticket.")
     registration_ids = fields.One2many('event.registration', 'sale_order_line_id', string="Registrations")
 
+    @api.constrains('event_id', 'event_ticket_id', 'product_id')
+    def _check_event_registration_ticket(self):
+        for so_line in self:
+            if so_line.product_id.detailed_type == "event" and (not so_line.event_id or not so_line.event_ticket_id):
+                raise ValueError(_("A sale order line linked to a registration need an event and a ticket."))
```

Comment:
Also can't be a ValueError, needs to be a ValidationError, otherwise it's not very user-friendly in the display:

![image](https://github.com/odoo/odoo/assets/41282536/0d6f3290-dea4-4352-a736-65d6672477eb)